components/director/internal/open_resource_discovery/validation.go

```diff
@@ -1152,9 +1171,10 @@ func validateExtensibleInnerFields(el gjson.Result) error {
 
 	descriptionProperty := el.Get("description")
 	descriptionValue, ok := descriptionProperty.Value().(string)
+	validLength := len(descriptionValue) >= MinDescriptionLength && len(descriptionValue) <= MaxDescriptionLength
 
-	if supportedProperty.Exists() && (supportedValue == "manual" || supportedValue == "automatic") && (descriptionValue == "" || !ok) {
-		return errors.New("if supported is either 'manual' or 'automatic', description should be provided")
+	if supportedProperty.Exists() && (supportedValue == "manual" || supportedValue == "automatic") && (descriptionValue == "" || !ok || !validLength) {
```

Comment:
`descriptionValue == ""` is redundant as it will has a len of 0 which is not validLength.

Same for `!ok` as if the ok is false the descriptionValue would be empty string which will also has a len of 0.